Source/SwiftLintBuiltInRules/Rules/Lint/UnneededOverrideRule.swift

```diff
@@ -0,0 +1,89 @@
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+struct UnneededOverrideRule: ConfigurationProviderRule, SwiftSyntaxRule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "unneeded_override",
+        name: "Remove empty overridden functions",
+        description: "Remove overridden functions that don't do anything except call their super",
+        kind: .lint,
+        nonTriggeringExamples: UnneededOverrideRuleExamples.nonTriggeringExamples,
+        triggeringExamples: UnneededOverrideRuleExamples.triggeringExamples
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private func simplify(_ node: SyntaxProtocol) -> ExprSyntaxProtocol? {
+    if let expr = node.as(AwaitExprSyntax.self) {
+        return expr.expression
+    } else if let expr = node.as(TryExprSyntax.self) {
+        // Assume using try! / try? changes behavior
+        if expr.questionOrExclamationMark != nil {
+            return nil
+        }
+
+        return expr.expression
+    } else if let expr = node.as(FunctionCallExprSyntax.self) {
+        return expr
+    } else if let stmt = node.as(ReturnStmtSyntax.self) {
+        return stmt.expression
+    }
+
+    return nil
+}
+
+private final class Visitor: ViolationsSyntaxVisitor {
+    override func visitPost(_ node: FunctionDeclSyntax) {
+        guard let modifiers = node.modifiers, modifiers.contains(where: { $0.name.text == "override" }),
+            node.body?.statements.count == 1, let statement = node.body?.statements.first else {
+            return
+        }
+
+        // Assume having @available changes behavior
+        if node.attributes.contains(attributeNamed: "available") {
+            return
+        }
+
+        // Extract the function call from other expressions like try / await / return.
+        // If this returns a non-super calling function that will get filtered out later
+        var syntax: ExprSyntaxProtocol? = simplify(statement.item)
+        while let nestedSyntax = syntax {
+            if nestedSyntax.as(FunctionCallExprSyntax.self) != nil {
+                break
+            }
+
+            syntax = simplify(nestedSyntax)
+        }
```

Comment:
Maybe extract this into a function to avoid casting to `FunctionCallExprSyntax` twice.